moxy/org.eclipse.persistence.moxy/src/main/java/org/eclipse/persistence/jaxb/XMLBindingContextFactory.java

```diff
@@ -28,14 +30,43 @@ public XMLBindingContextFactory() {
         //no-op
     }
 
+    @FunctionalInterface
+    private interface ContextSupplier {
```

Comment:
I'm not sure I like this. Why to complicate simple `Map.getOrDefault` from passed in options?

Reply:
I could not use default Supplier interface because of Exceptios handling. This code is prepared for adding more factory types. That's why I made it a bit more complicated.

Reply:
it's complicated solution for a very simple problem. I'd rather avoid it.

Reply:
OK, rewritten to stupid simple form. :)